### ISSUE TITLE
Weird syntax highlight issue with Odin language map[string]bool

### ISSUE THREAD

**Issue by Akselmo** [helix-editor/helix#11520]
### Summary

Hi, I wasn't exactly sure where to report this, since https://github.com/tree-sitter-grammars/tree-sitter-odin mentioned following:

> If your issue is related to a bug in your editor-experience because your editor leverages tree-sitter and this parser, then it is likely your issue does NOT belong here and belongs in the relevant editor's repository.

# The bug

If I create a map with strings as keys, the coloring seems to get confused.

Best shown with a screenshot:

![image](https://github.com/user-attachments/assets/8c6132ed-fc48-42d8-8959-11b06bec9564)

The first line, `MOVE_FRONT`, is correct. Any subsequent lines look wrong. 

Tree-sitter scopes seem to error out?

![image](https://github.com/user-attachments/assets/4f927cf9-ca46-4aad-a9d7-748ae2265ff9)

![image](https://github.com/user-attachments/assets/f240bed7-9ca2-4996-8c85-6bfb66f78097)

### Reproduction Steps

 

I tried this:

1. `hx`
2. Paste following snippet
```odin
package test
import rl "vendor:raylib"

WeirdMap := map[string]bool \
{
	"MOVE_FRONT" = rl.IsKeyDown(cast(rl.KeyboardKey)settings.Values.keyMoveForward),
	"MOVE_BACK"  = rl.IsKeyDown(cast(rl.KeyboardKey)settings.Values.keyMoveBackward),
	"MOVE_RIGHT" = rl.IsKeyDown(cast(rl.KeyboardKey)settings.Values.keyMoveRight),
	"MOVE_LEFT"  = rl.IsKeyDown(cast(rl.KeyboardKey)settings.Values.keyMoveLeft),
}
```

I expected this to happen:
Syntax highlighting for string would be correct, like on the first line.

Instead, this happened:
Seems helix thinks for some reason that the next line with `MOVE_BACK` with the first `"` is not start of string?

### Helix log

<details><summary>~/.cache/helix/helix.log</summary>

```
please provide a copy of `~/.cache/helix/helix.log` here if possible, you may need to redact some of the lines
```

</details>

### Platform

Linux

### Terminal Emulator

konsole 24.11.70

### Installation Method

fedora packages

### Helix Version

helix 24.7 (079f5442)

**Comment by archseer** [comment]
No, this is definitely a parsing issue that has to be fixed by the grammar upstream. (I also can't find that quote anywhere)

**Comment by Akselmo** [comment]
I saw the quote when making a bug report. Anyhow, thanks for clarification, will make a bug upstream :)

**Comment by leo-unglaub** [comment]
Just for reference, this is not just linked to the `odin` language. This happens also in `php` and other languages, most of the time if the letter `M` is involved.

![Screenshot from 2024-08-21 12-56-43](https://github.com/user-attachments/assets/ec610868-b188-4147-a1f5-2461f6eaf063)

**Comment by the-mikedavis** [comment]
That's unrelated - it's because the `k` is lowercase so it's not highlighted as a constant. That's up to php's queries in `runtime/queries/php/highlights.scm`
